internal/ai/conversation.go

```diff
@@ -157,6 +178,115 @@ func getLastCacheControl(content []anthropic.ContentBlockParamUnion) (*anthropic
+	if len(cc.Messages) <= numFirstMessagesToPreserve {
+		return nil
+	}
+
+	// Generate summary using AI (this will add a summary request/response to the conversation)
+	summaryResponse, err := cc.generateConversationSummary(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to generate conversation summary: %w", err)
+	}
+
+	// Create conversation turns that properly represent the summary exchange
+	// First turn: User asks for summary + Assistant provides the summary
+	summaryRequestTurn := conversationTurn{
+		UserMessage: anthropic.NewUserMessage(anthropic.NewTextBlock("Please respond with the summary you generated earlier.")),
+		Response:    summaryResponse,
+	}
+	
+	// Second turn: User asks to resume work based on the summary
+	resumeRequestTurn := conversationTurn{
+		UserMessage: anthropic.NewUserMessage(anthropic.NewTextBlock("Please resume working on this task based on your summary.")),
+		// No response yet - this will be filled in by the next actual conversation turn
+	}
+
+	// Reconstruct the conversation: preserved first messages + summary exchange + resume request
+	newMessages := []conversationTurn{}
+	
+	// Add preserved first messages
+	newMessages = append(newMessages, cc.Messages[:numFirstMessagesToPreserve]...)
+	
+	// Add summary conversation turns
+	newMessages = append(newMessages, summaryRequestTurn, resumeRequestTurn)
+
+	// Update the conversation
+	originalMessageCount := len(cc.Messages)
+	cc.Messages = newMessages
+
+	log.Printf("Conversation summarized: %d messages -> %d messages", 
+		originalMessageCount, len(cc.Messages))
+
+	return nil
+}
+
+// generateConversationSummary creates a summary of the conversation using AI
+func (cc *Conversation) generateConversationSummary(ctx context.Context) (*anthropic.Message, error) {
+	// Build a summary request that will be sent as part of the current conversation
+	var summaryPrompt strings.Builder
+	summaryPrompt.WriteString("Please summarize all of the work you have done so far. Focus on:\n")
+	summaryPrompt.WriteString("1. Key decisions and changes made\n")
+	summaryPrompt.WriteString("2. Current state of the codebase\n")
+	summaryPrompt.WriteString("3. Any important context for continuing the work\n")
+	summaryPrompt.WriteString("4. Tools used and their outcomes\n\n")
+	summaryPrompt.WriteString("Please provide a comprehensive but concise summary that captures all important information needed to continue working effectively. ")
+	summaryPrompt.WriteString("There's no need to include the system prompt or initial repository information in your summary - focus on the actual work and changes made during our conversation.")
+
+	// Send summary request without caching since we're throwing away conversation history
+	response, err := cc.sendMessage(ctx, false, anthropic.NewTextBlock(summaryPrompt.String()))
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate summary: %w", err)
+	}
+
+	// Return the complete response message to preserve proper authorship
```

Comment:
This comment explains a detail of this conversation that won't be relevant to future readers. Let's remove the mention of preserving authorship